plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestDateCoercer.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.coercions;
+
+import io.trino.spi.type.DateType;
+import io.trino.spi.type.VarcharType;
+import org.testng.annotations.Test;
+
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.trino.plugin.hive.HiveTimestampPrecision.NANOSECONDS;
+import static io.trino.plugin.hive.coercions.TestTimestampCoercer.assertCoercions;
+
+public class TestDateCoercer
+{
+    @Test
+    public void testValidUnboundedVarcharToDate()
+    {
+        assertCoercions(VarcharType.createUnboundedVarcharType(), utf8Slice("2023-04-13"), DateType.DATE, 19460L, NANOSECONDS);
+    }
+
+    @Test
+    public void testValidBounded10VarcharToDate()
+    {
+        assertCoercions(VarcharType.createVarcharType(10), utf8Slice("2023-04-13"), DateType.DATE, 19460L, NANOSECONDS);
+    }
+
+    @Test
+    public void testValidUnbounded10VarcharWithLongYearToDate()
+    {
+        assertCoercions(VarcharType.createUnboundedVarcharType(), utf8Slice("+10000-04-13"), DateType.DATE, 2933000L, NANOSECONDS);
```

Comment:
Is it possible to use DataProvider to play with some historical dates and futuristic dates ?

Reply:
If its unbounded should mention 10 as a part of method name ?